Workload-Security/Integration/aws-python-amazon-inspector-integration-v2/templates/c1ws-inspector-findings-template.yaml

```diff
@@ -0,0 +1,702 @@
+    Properties:
+      BucketName: !Ref BucketName
+  LambdaSecretAcessPolicy: 
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Sid: VisualEditor0
+          Effect: Allow
+          Action:
+          - "secretsmanager:GetSecretValue"
+          - "secretsmanager:DescribeSecret"
+          - "sts:AssumeRole"
+          Resource: !Ref ApiKeySecret
+  IpsRuleAutoAssignLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+        Version: 2012-10-17
+      ManagedPolicyArns:
+          - arn:aws:iam::aws:policy/AmazonVPCCrossAccountNetworkInterfaceOperations
+          - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+          - arn:aws:iam::aws:policy/AmazonInspector2ReadOnlyAccess
+          - arn:aws:iam::aws:policy/CloudWatchLambdaInsightsExecutionRolePolicy
+          - !Ref LambdaSecretAcessPolicy
+      Path: /
+      Policies:
+        - PolicyName: SendCommandPolicy
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - ssm:SendCommand
+                Resource: "*"
+        - PolicyName: SnsNotificationPolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action: sns:Publish
+                Resource: !Ref SNSTopic
+        - PolicyName: S3WritePolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action: s3:PutObject
+                Resource: !Sub 'arn:aws:s3:::${BucketName}/*'
+  IpsRuleAutoAssignLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      Handler: index.lambda_handler
+      MemorySize: 128
+      Runtime: python3.9
+      Timeout: 60
+      Environment:
+        Variables:
+          cloudoneregion: !Ref CloudOneRegion
+          c1_api_key: !Ref ApiKeySecret
+          bucket_name: !Ref BucketName
+          topic_arn: !Ref SNSTopic
+          csv_filename: !Sub "${AWS::Region}-ips_rule_auto_assign_report.csv"
+          aws_region: !Ref AWS::Region
+      Role: !GetAtt IpsRuleAutoAssignLambdaRole.Arn
+      Code:
+        ZipFile: |
+            import os
+            import csv
+            import json
+            from copy import deepcopy
+            from urllib.request import Request, urlopen
+            import boto3
+
+
+            API_VERSION = "v1"
+            TOPIC_ARN = os.environ.get('topic_arn')
+            AWS_REGION = os.environ.get("aws_region")
+            BUCKET_NAME = os.environ.get("bucket_name")
+            FILE_NAME = os.environ.get("csv_filename")
+            C1_API_KEY = os.environ.get("c1_api_key")
```

Comment:
I suggest using a boto3 secrets manager client rather than an environment variable given you already created a secret in line 71.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager.html